bot/seasons/valentines/be_my_valentine.py

```diff
@@ -0,0 +1,235 @@
+
+        syntax : .bemyvalentine secret [user](optional) [p/poem/c/compliment/or you can type your own valentine message]
+        (optional)
+
+        example : .bemyvalentine secret (sends valentine as a poem or a compliment to a random user in DM making you
+        anonymous)
+        example : .bemyvalentine secret Iceman#6508 p (sends a poem to Iceman in DM making you anonymous)
+        example : .bemyvalentine secret Iceman#6508 Hey I love you, wanna hang around ? (sends the custom message to
+        Iceman in DM making you anonymous)
+        """
+
+        if ctx.guild is not None:
+            # This command is only DM specific
+            msg = "You are not supposed to use this command in the server, DM the command to the bot."
+            return await ctx.send(msg)
+
+        if user:
+            if Lovefest.role_id not in [role.id for role in user.roles]:
+                message = f"You cannot send a valentine to {user} as he/she does not have the lovefest role!"
+                return await ctx.send(message)
+
+        if user == ctx.author:
+            # Well a user cant valentine himself/herself.
+            return await ctx.send('Come on dude, you cant send a valentine to yourself :expressionless:')
+
+        guild = self.bot.get_guild(id=Client.guild)
+        emoji_1, emoji_2 = self.random_emoji()
+        lovefest_role = discord.utils.get(guild.roles, id=Lovefest.role_id)
+        valentine, title = self.valentine_check(valentine_type)
+
+        if user is None:
+            author = ctx.author
+            user = self.random_user(author, lovefest_role.members)
+            if user is None:
+                return await ctx.send("There are no users avilable to whome your valentine can be sent.")
+
+        embed = discord.Embed(
+            title=f'{emoji_1}{title} {user.display_name}{emoji_2}',
+            description=f'{valentine} \n **{emoji_2}From anonymous{emoji_1}**',
+            color=Colours.pink
+        )
+        await ctx.author.send(f"Your message has been sent to {user}")
+        await user.send(embed=embed)
+
+    def valentine_check(self, valentine_type):
+        if valentine_type is None:
+            valentine, title = self.random_valentine()
+
+        elif valentine_type.lower() in ['p', 'poem']:
+            valentine = self.valentine_poem()
+            title = 'A poem dedicated to'
+
+        elif valentine_type.lower() in ['c', 'compliment']:
+            valentine = self.valentine_compliment()
+            title = 'A compliment for'
+
+        else:
+            # in this case, the user decides to type his own valentine.
+            valentine = valentine_type
+            title = 'A message for'
+        return valentine, title
+
+    @staticmethod
+    def random_user(author, members):
+        # just making sure that the random does not pick up the same user(ctx.author)
+
+        if members.__len__() != 0:
+            if author in members:
+                members.remove(author)
```

Comment:
The length check is done before removing the original author, so if the author is the only person with the role, it'll pass the if check, but then return an empty list, later raising an exception since `random.choice` fails with empty collections.

Check membership first, and then check length after removing them if they're present. This ensures it'll be seen as no valid members to receive it, and avoids `random.choice` failing.

```py
    @staticmethod
    def random_user(author, members):
        # just making sure that the random does not pick up the same user(ctx.author)
        if author in members:
            members.remove(author)
        if len(members) != 0:
            user = random.choice(members)
        else:
            user = None
        return user
```